scripts/ecs/appsignals/setup-ecs-demo.sh

```diff
@@ -0,0 +1,1124 @@
+      --cluster $CLUSTER > /dev/null
+
+  sleep 120
+
+  echo "Finished deploying admin server!"
+
+}
+
+function run_api_gateway() {
+  api_gateway_target_group_arn=$(aws elbv2 create-target-group \
+      --name api-gateway-target-group \
+      --protocol HTTP \
+      --port 8080 \
+      --vpc-id $VPC_ID \
+      --target-type ip \
+      --query "TargetGroups[0].TargetGroupArn" \
+      --output text)
+
+  aws elbv2 modify-target-group \
+      --target-group-arn $api_gateway_target_group_arn \
+      --health-check-protocol HTTP \
+      --health-check-path "/" \
+      --health-check-interval-seconds 240 \
+      --health-check-timeout-seconds 60 \
+      --healthy-threshold-count 5 \
+      --unhealthy-threshold-count 2 > /dev/null
+
+  aws elbv2 create-listener \
+    --load-balancer-arn $LOAD_BALANCER_ARN \
+    --protocol HTTP \
+    --port 8080 \
+    --default-actions Type=forward,TargetGroupArn=$api_gateway_target_group_arn > /dev/null
+
+  adot_java_image_tag=$(curl -s -I -L 'https://github.com/aws-observability/aws-otel-java-instrumentation/releases/latest' | grep -i Location | awk -F'/tag/' '{print $2}' | tr -d '\r')
+  adot_java_image="public.ecr.aws/aws-observability/adot-autoinstrumentation-java:$adot_java_image_tag"
+  sed -i '' "s|\"adot-java-image\"|\"${adot_java_image}\"|" ./sample-app/task-definitions/spring-petclinic-api-gateway.json
+  sed -i '' "s|\"api_gateway_ip\"|\"${LOAD_BALANCER_DNS}\"|" ./sample-app/task-definitions/spring-petclinic-api-gateway.json
+  sed -i '' "s|\"api-gateway-image\"|\"${ECR_IMAGE_PREFIX}/springcommunity/spring-petclinic-api-gateway\"|" ./sample-app/task-definitions/spring-petclinic-api-gateway.json
+  sed -i '' "s|us-west-2|${REGION}|g" ./sample-app/task-definitions/spring-petclinic-api-gateway.json
+  sed -i '' "s|000111222333|${ACCOUNT_ID}|g" ./sample-app/task-definitions/spring-petclinic-api-gateway.json
+
+  aws ecs register-task-definition --cli-input-json file://sample-app/task-definitions/spring-petclinic-api-gateway.json > /dev/null
+
+  aws ecs create-service \
+      --service-name api-gateway \
+      --task-definition api-gateway \
+      --load-balancers targetGroupArn=$api_gateway_target_group_arn,containerName=api-gateway,containerPort=8080 \
+      --desired-count 1 \
+      --launch-type FARGATE \
+      --network-configuration "awsvpcConfiguration={subnets=[$(echo $SUBNET_IDS | tr -s ' ' ',')],securityGroups=[$SECURITY_GROUP_ID],assignPublicIp=ENABLED}" \
+      --cluster $CLUSTER > /dev/null
+
+  echo "Waiting for the Frontened server to be accessible..."
+  sleep 240
+
+  echo "Frontened server is now accessible!"
+
+}
+
+function run_vets_service() {
+  namespace_id=$(aws servicediscovery list-namespaces --query "Namespaces[?Name=='ecs-petclinic'].Id" --output text)
+
+  service_discovery_id=$(aws servicediscovery create-service \
```

Comment:
Why do we need service discovery for vets-service? Same question applies to the other pet-clinic services.

Reply:
In petclinic, each service registered with eureka with it's IP and port, then it expose health check endpoint at: `http://<service_ip>:<port>/actuator/health`, where the service will automatically deregister from Eureka if its `actuator/health` check fails. 

In ECS, the app cannot be reachable from either the instance IP (as each task has it's own IP) nor the container name. We use service discovery to expose the app with <`service_name.namespace>` format. Then the `http://<service_name.namespace>:<port>/actuator/health` can be accessible. For example vets-service, the endpoint is `http://vets-service.ecs-petclinic:<port>/actuator/health`

Reply:
Added on the top of the file.